application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -4603,6 +4603,42 @@ input TransferPreferencesInput {
   slack: Duration
 }
 
+"A collection of selectors for what routes/agencies should be included in or excluded from the search."
+input TransitFilterInput {
+  """
+  A list of selectors for what routes/agencies should be allowed during search.
+    
+  In order to be excluded a route/agency has to match with at least one selector.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  exclude: [TransitFilterSelectInput!]
+  """
+  A list of selectors for what routes/agencies should be excluded during the search.
+    
+  If route/agency matches with at least one selector it will be included.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  include: [TransitFilterSelectInput!]
+}
+
+"""
+A list of selectors for filter including or excluding entities from the routing results. Null
+means everything is allowed to be returned and empty lists a not allowed.
+
+An entity will match with selectors if it matches with all non-empty lists.
+
+The `include` is always applied first, then `exclude`. If only `exclude` is present, the exclude is
+applied to the existing set of results.
+"""
+input TransitFilterSelectInput {
+  "Set of ids for agencies that should be included in/excluded from search"
+  agencies: [String!]
+  "Set of ids for routes that should be included in/excluded from search"
+  routes: [String!]
```

Comment:
Usually we include an example of what the id is (for example "i.e. value of field `gtfsId` (ID format is `FeedId:StopId`)")